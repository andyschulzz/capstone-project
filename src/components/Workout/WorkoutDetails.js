import React from 'react'
import Details from './Details.js'
import { GreenButton } from '../common/Button'
import { useForm } from 'react-hook-form'
import { useHistory } from 'react-router-dom'
import * as S from './WorkoutDetails.styles'
import PropTypes from 'prop-types'

WorkoutDetails.propTypes = {
  selectedWorkouts: PropTypes.array.isRequired,
  handleWorkoutSubmit: PropTypes.func.isRequired,
}

export default function WorkoutDetails({
  selectedWorkouts,
  handleWorkoutSubmit,
}) {
  const { register, handleSubmit } = useForm()
  const history = useHistory()

  return (
    <S.Form
      data-testid="details-form"
      id="details"
      onSubmit={handleSubmit(handleDetails)}
    >
      <S.ButtonWrapper>
        <h3>Workout Volume</h3>
        <GreenButton inactive mla>
          Save
        </GreenButton>
      </S.ButtonWrapper>
      {selectedWorkouts.map((workout, index) => {
        return (
          <Details
            register={register}
            key={workout.id}
            {...workout}
            index={index}
          />
        )
      })}
    </S.Form>
  )

  function handleDetails(data) {
    handleWorkoutSubmit(data)
    history.push('/workouts')
  }
}
