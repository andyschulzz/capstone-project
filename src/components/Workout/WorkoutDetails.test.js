import React from 'react'
import { render, cleanup } from '@testing-library/react'
import WorkoutDetails from './WorkoutDetails.js'
import { MemoryRouter as Router } from 'react-router-dom'

afterEach(cleanup)

const handleWorkoutAddMock = jest.fn()
const workoutExercisesMock = [
  {
    id: 1,
    name: 'Test',
    type: 'Test',
    instructions: 'Test',
    title: 'Test',
  },
]

describe('<WorkoutDetails />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <Router>
        <WorkoutDetails
          workoutExercises={workoutExercisesMock}
          handleWorkoutAdd={handleWorkoutAddMock}
        />
      </Router>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('Form should not be valid if input is missing', () => {
    const { getByTestId } = render(
      <Router>
        <WorkoutDetails
          workoutExercises={workoutExercisesMock}
          handleWorkoutAdd={handleWorkoutAddMock}
        />
      </Router>
    )
    const inputNode = getByTestId('details-form')
    expect(inputNode).not.toBeValid()
  })
})
