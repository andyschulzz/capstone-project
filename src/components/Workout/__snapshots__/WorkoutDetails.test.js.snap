// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkoutDetails /> should render correctly 1`] = `
<DocumentFragment>
  .c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 input {
  width: 48px;
  margin-top: 5px;
  padding-top: 3px;
  text-align: center;
  font-size: 1.2rem;
  border: 1px solid #ddd;
  border-radius: 5px;
  outline: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c4 p {
  margin: 0;
  text-align: center;
  font-weight: 300;
  color: #888;
  text-transform: uppercase;
}

.c4 span {
  margin: 0 10px;
}

.c3 {
  font-weight: 400;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: none;
  border: none;
  outline: none;
  border-radius: 5px;
  margin-left: auto;
  padding: 5px 15px;
  font-size: 1rem;
  height: 36px;
  cursor: pointer;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: hsla(120,50%,50%,0.2);
}

.c2 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: white;
}

.c0 {
  padding: 0 40px 30px;
  overflow-y: auto;
  -webkit-scroll-behavior: smooth;
  -moz-scroll-behavior: smooth;
  -ms-scroll-behavior: smooth;
  scroll-behavior: smooth;
}

.c0 h3:first-child {
  margin: 0;
}

.c0 h3 h3 {
  margin-top: 0;
}

.c0:valid button {
  background: hsl(120,50%,50%);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  background: #fff;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 40px 0 30px;
}

<form
    class="c0"
    data-testid="details-form"
    id="details"
  >
    <div
      class="c1"
    >
      <h3>
        Workout Volume
      </h3>
      <button
        class="Button-sc-18sujpt-0 c2"
      >
        Save
      </button>
    </div>
    <h3
      class="c3"
    >
      Test
    </h3>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <p>
          sets
        </p>
        <input
          min="1"
          name="sets[0]"
          pattern="[0-9]*"
          required=""
          type="text"
        />
      </div>
      <span>
        ×
      </span>
      <div
        class="c5"
      >
        <p>
          reps
        </p>
        <input
          min="1"
          name="reps[0]"
          pattern="[0-9]*"
          required=""
          type="text"
        />
      </div>
      <span>
        ×
      </span>
      <div
        class="c5"
      >
        <p>
          weight
        </p>
        <input
          name="weight[0]"
          pattern="[0-9]*"
          type="text"
        />
      </div>
    </div>
  </form>
</DocumentFragment>
`;
